src/fromager/sources.py

```diff
@@ -414,7 +416,7 @@ def build_sdist(
         extra_environ=extra_environ,
         req=req,
         version=version,
-        sdist_root_dir=sdist_root_dir,
+        sdist_root_dir=ctx.settings.build_dir(req.name, sdist_root_dir),
```

Comment:
I'm trying to think through if there are any cases where an override plugin would need to know both the root of the source tree and the directory where it should do the build. For example, the pyarrow build? Maybe we just don't use the configuration option in that case, though.

What do you think?

Reply:
Yeah, i think if the user needs both then they shouldn't use this option and use plugins instead

Reply:
or maybe for now i don't use this option for `build_wheel` which will allow us to use this feature for the pyarrow builds?

Reply:
I think we'd want to be consistent. After prepare_source, all overrides should either get both paths or only one of them.

Reply:
i think then lets just return the build dir

Reply:
Yeah, I think that works. For the pyarrow case, we'll need to compute the path relative to that directory to build the stuff outside of the python directory, but that's OK.